lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -351,30 +348,34 @@ private SignedJWT validateHmrcInheritedIdentity(
         }
 
         // Validate VC in claim is valid
-        var signedInheritedIdentityJWT = SignedJWT.parse(inheritedIdentityJwtList.get(0));
+        var inheritedIdentityJwt = SignedJWT.parse(inheritedIdentityJwtList.get(0));
         var inheritedIdentityCriConfig = configService.getCriConfig(HMRC_MIGRATION_CRI);
 
         // The HMRC inherited identity VC will contain an HMRC-specific pairwise identifier
         // rather than our internal user id, so we cannot validate it against the OAuth user id.
         // Instead, SPOT will validate this when generating an identity bundle.
-        var inheritedUserId = signedInheritedIdentityJWT.getJWTClaimsSet().getSubject();
-        verifiableCredentialJwtValidator.validate(
-                signedInheritedIdentityJWT, inheritedIdentityCriConfig, inheritedUserId);
+        var inheritedUserId = inheritedIdentityJwt.getJWTClaimsSet().getSubject();
+        var inheritedIdentityVc =
+                verifiableCredentialJwtValidator.validate(
+                        inheritedIdentityJwt.serialize(),
+                        HMRC_MIGRATION_CRI,
+                        inheritedIdentityCriConfig.getParsedSigningKey(),
+                        inheritedIdentityCriConfig.getComponentId(),
+                        inheritedUserId);
         LOGGER.info(LogHelper.buildLogMessage("Migration VC successfully validated"));
 
-        return signedInheritedIdentityJWT;
+        return inheritedIdentityVc;
     }
 
     private void storeHmrcInheritedIdentity(
             String userId,
             JWTClaimsSet claimsSet,
             IpvSessionItem ipvSessionItem,
-            SignedJWT signedInheritedIdentityJWT)
+            VerifiableCredential inheritedIdentityVc)
             throws RecoverableJarValidationException, ParseException {
         try {
-            verifiableCredentialService.persistUserCredentials(
-                    signedInheritedIdentityJWT, HMRC_MIGRATION_CRI, userId);
-            ipvSessionItem.addVcReceivedThisSession(signedInheritedIdentityJWT.serialize());
+            verifiableCredentialService.persistUserCredentials(inheritedIdentityVc, userId);
```

Comment:
no longer needs to reference the criId since it's baked into the VerifiableCredential